rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -681,15 +703,18 @@ void RSocketStateMachine::handleUnknownStream(
   // TODO: comparing string versions is odd because from version
   // 10.0 the lexicographic comparison doesn't work
   // we should change the version to struct
-  if (frameSerializer_->protocolVersion() > ProtocolVersion{0, 0} &&
-      !streamsFactory_.registerNewPeerStreamId(streamId)) {
-    return;
+  if(frameType != FrameType::PAYLOAD) {
```

Comment:
It will register the stream ID for the incoming REQUEST, and then drop the rest of the fragments coming in via PAYLOAD frames later. We only want to register a new stream ID for non-Payload types, in that case.